tests/test_verilog/test_from_file.py

```diff
@@ -118,99 +121,98 @@ def definition(io):
+#     [foo, bar] = m.define_from_verilog("""
+# module foo(input I, output O);
+#     assign O = I;
+# endmodule
+
+# module bar(input I, output O);
+#     foo foo_inst(I, O);
+# endmodule""")
+#     top = m.DefineCircuit("top", "I", m.In(m.Bit), "O", m.Out(m.Bit))
+#     bar_inst = bar()
+#     m.wire(top.I, bar_inst.I)
+#     m.wire(bar_inst.O, top.O)
+#     m.EndDefine()
+#     FILENAME = "test_verilog_dependency_top"
+#     m.compile(f"build/{FILENAME}", top, output="coreir")
+#     assert m.testing.check_files_equal(__file__, f"build/{FILENAME}.json",
+#                                        f"gold/{FILENAME}.json")
+
+
+# def test_verilog_dependency_out_of_order():
+#     [foo, bar] = m.define_from_verilog("""
+# module bar(input I, output O);
+#     foo foo_inst(I, O);
+# endmodule
+
+# module foo(input I, output O);
+#     assign O = I;
+# endmodule""")
+#     top = m.DefineCircuit("top", "I", m.In(m.Bit), "O", m.Out(m.Bit))
+#     bar_inst = bar()
+#     m.wire(top.I, bar_inst.I)
+#     m.wire(bar_inst.O, top.O)
+#     m.EndDefine()
+#     FILENAME = "test_verilog_dependency_top"
+#     m.compile(f"build/{FILENAME}", top, output="coreir")
+#     assert m.testing.check_files_equal(__file__, f"build/{FILENAME}.json",
+#                                        f"gold/{FILENAME}.json")
+
+
+# def test_from_verilog_external_modules():
+#     [foo] = m.define_from_verilog("""
+# module foo(input I, output O);
+#     assign O = I;
+# endmodule""")
+#     [bar] = m.define_from_verilog("""
+# module bar(input I, output O);
+#     foo foo_inst(I, O);
+# endmodule""", external_modules={"foo": foo})
+#     top = m.DefineCircuit("top", "I", m.In(m.Bit), "O", m.Out(m.Bit))
+#     bar_inst = bar()
+#     m.wire(top.I, bar_inst.I)
+#     m.wire(bar_inst.O, top.O)
+#     m.EndDefine()
+#     FILENAME = "test_verilog_dependency_top"
+#     m.compile(f"build/{FILENAME}", top, output="coreir")
+#     assert m.testing.check_files_equal(__file__, f"build/{FILENAME}.json",
+#                                        f"gold/{FILENAME}.json")
+
+
+# def test_from_verilog_external_modules_missing():
+#     with pytest.raises(Exception) as pytest_e:
+#         m.define_from_verilog("""
+# module bar(input I, output O);
+#     foo foo_inst(I, O);
+# endmodule""")
+#         assert False
+#     assert pytest_e.type is KeyError
+#     assert pytest_e.value.args == ("foo",)
+
+
+# def test_from_verilog_external_modules_duplicate():
+#     with pytest.raises(MultipleModuleDeclarationError) as pytest_e:
+#         [foo] = m.define_from_verilog("""
+# module foo(input I, output O);
+#     assign O = I;
+# endmodule""")
+#         _ = m.define_from_verilog("""
+# module foo(input I, output O);
+#     assign O = I;
+# endmodule
+
+# module bar(input I, output O);
+#     foo foo_inst(I, O);
+# endmodule""", external_modules={"foo": foo})
+#         assert False
+#     assert pytest_e.type is MultipleModuleDeclarationError
+#     assert pytest_e.value.args == ("foo",)
```

Comment:
Should this be added back in?

Reply:
Yes, planning to refactor the tests a bit too.

Reply:
Actually, I misspoke. This change gets rid of the functionality these tests were originally written for. This changes gets rid of anything to do with `shallow` or `external_modules`.

I think we want to move towards importing verilog only for the sake of getting the interface; we shouldn't be passing the verilog down to CoreIR. It should be linked in otherwise.